gufe/storage/storagemanager.py

```diff
@@ -0,0 +1,108 @@
+from contextlib import contextmanager
+from pathlib import Path
+from typing import Literal
+
+from .externalresource import ExternalStorage
+
+
+class StorageManager:
+    """Manage storage operations for files in a DAG.
+
+    This class provides a context manager for working with storage systems,
+    allowing registration, loading, and transfer of files between scratch
+    storage and external storage.
```

Comment:
```suggestion
    This class provides a context manager for working with storage systems,
    allowing registration, loading, and transfer of files between scratch
    directory and external storage.
```

Trying to make the docs super clear about this.